### ISSUE TITLE
Connection Timeouts

### ISSUE THREAD

**Issue by bfrog** [sfackler/rust-postgres#246]
Are there any timeouts on waiting for a connection to send or receive results back? I couldn't find any and I'm having an issue where when the postgres server is restarted my http service seems to get stuck somewhere performing queries.

It would be nice to be able to set timeouts on sending and receiving results if not possible at the moment. If it is and I missed it apologies in advance!

**Comment by sfackler** [comment]
There are not any timeouts currently (other than when reading notifications). It seems reasonable to add some methods allowing you to set the socket read and write timeouts or TCP keepalive though!

**Comment by william20111** [comment]
Yes this would be a pretty neat feature something that python postgres module has and i use for monitoring checks etc.

**Comment by sfackler** [comment]
#286 adds connect timeouts.

**Comment by william20111** [comment]
your awesome! look forward to this landing.
